pipeline/console/environments.py

```diff
@@ -18,14 +18,15 @@
 environment_re_pattern = re.compile(r"^[0-9a-zA-Z]+[0-9a-zA-Z\_\-]+[0-9a-zA-Z]+$")
 
 
-def _get_environment(name_or_id: str) -> EnvironmentGet:
-
+def _get_environment(name_or_id: str, by_name=False) -> EnvironmentGet:
     remote_service = PipelineCloud(verbose=False)
     remote_service.authenticate()
 
-    environment_information = EnvironmentGet.parse_obj(
-        remote_service._get(f"/v2/environments/{name_or_id}")
-    )
+    url_config = ["/v2/environments", name_or_id]
+    separator = "/" if not by_name else "/by-name/"
+    url = separator.join(url_config)
```

Comment:
This hurt my brain a little 😄 how about

```suggestion
    url = f"/v2/environments/by-name/{name_or_id}" if by_name else f"/v2/environments/{name_or_id}"
```